servant-pt1.cabal

```diff
@@ -38,20 +38,32 @@ library
                      , say
                      , servant >= 0.15
                      , servant-server
+                     , stm
                      , string-conversions >= 0.4.0.1
                      , text >= 1.2.3.1
                      , text-show
                      , time >= 1.8.0.2
                      , wai
                      , warp
   hs-source-dirs:      src
+  exposed-modules:     Database
+                       Server
+                       Types
   default-language:    Haskell2010
+  other-extensions:    QuasiQuotes
+                       TemplateHaskell
+  default-extensions:  DataKinds
```

Comment:
If you're writing an application (as opposed to a library you upload to Hackage), it is often nice to throw the standard extensions into the cabal file so you don't have to write them at the top of each module.

Here's the extension list we use at work:

```yaml
  - AutoDeriveTypeable
  - BangPatterns
  - BinaryLiterals
  - ConstraintKinds
  - DataKinds
  - DefaultSignatures
  - DeriveDataTypeable
  - DeriveFoldable
  - DeriveFunctor
  - DeriveGeneric
  - DeriveTraversable
  - DoAndIfThenElse
  - EmptyCase
  - EmptyDataDecls
  - ExistentialQuantification
  - ExplicitNamespaces
  - FlexibleContexts
  - FlexibleInstances
  - FunctionalDependencies
  - GADTs
  - GeneralizedNewtypeDeriving
  - InstanceSigs
  - KindSignatures
  - LambdaCase
  - MultiParamTypeClasses
  - MultiWayIf
  - NamedFieldPuns
  - NoImplicitPrelude
  - OverloadedStrings
  - PartialTypeSignatures
  - PatternGuards
  - PolyKinds
  - RankNTypes
  - RecordWildCards
  - ScopedTypeVariables
  - StandaloneDeriving
  - TupleSections
  - TypeFamilies
  - TypeSynonymInstances
  - ViewPatterns
```